test/integration/consul-container/libs/cluster/container.go

```diff
@@ -135,6 +136,13 @@ func NewConsulContainer(ctx context.Context, config Config, cluster *Cluster, po
 		if err != nil {
 			return nil, fmt.Errorf("error copying persistent data from %s: %w", config.ExternalDataDir, err)
 		}
+		// NOTE: make sure the new version can access the persistent data
+		// This is necessary for running on Linux
+		cmd := exec.Command("chmod", "-R", "777", tmpDirData)
+		err = cmd.Run()
+		if err != nil {
+			return nil, fmt.Errorf("error changing ownership of persistent data: %w", err)
```

Comment:
I changed this from a `panic` to an actual error in both places as the panic felt weird given this already had an error return as an option.